main.go

```diff
@@ -0,0 +1,138 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"math"
+	"os"
+)
+
+func Isletter(b byte) bool {
+	return b >= 65
+}
+func tobase10(s string, p int) (a int) {
+	c := 1
+
+	switch p > 0 {
+	case (p <= 10):
+		for i := len(s) - 1; i >= 0; i-- {
+			a = a + int((s[i])-48)*c
+			c = c * p
+		}
+
+	case p > 10:
+		for i := len(s) - 1; i >= 0; i-- {
+			if !Isletter(s[i]) {
+				a = a + int((s[i])-48)*c
```

Comment:
Please use runes instead of hardcoded numerical constants, e.g. `'0'` instead of 48